app/src/main/java/org/mozilla/focus/locale/screen/LanguageStorage.kt

```diff
@@ -0,0 +1,97 @@
+import android.util.Log
+import androidx.preference.PreferenceManager
+import org.mozilla.focus.R
+import org.mozilla.focus.locale.LocaleManager
+import java.util.Arrays
+
+class LanguageStorage(private val context: Context) {
+
+    /**
+     * Returns the current selected Language object or System default Language if nothing is selected
+     */
+    fun getSelectedLanguageTag(): Language {
+        val sharedConfig: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+        val languageTag = sharedConfig.getString(
+            context.resources.getString(R.string.pref_key_locale),
+            LOCALE_SYSTEM_DEFAULT
+        ) ?: LOCALE_SYSTEM_DEFAULT
+        for (language in getLanguages()) {
+            if (languageTag == language.tag) {
+                return language
+            }
+        }
+        return Language(context.getString(R.string.preference_language_systemdefault), LOCALE_SYSTEM_DEFAULT, 0)
+    }
+
+    /**
+     * Returns The full list of languages available.System default Language  will be the first item in the list .
+     */
+    fun getLanguages(): List<Language> {
+        val listLocaleNameAndTag = ArrayList<Language>()
+        val descriptors = getUsableLocales()
+        listLocaleNameAndTag.add(
+            Language(
+                context.getString(
+                    R.string.preference_language_systemdefault
+                ),
+                LOCALE_SYSTEM_DEFAULT, 0
+            )
+        )
+        descriptors.indices.forEach { i ->
+            val displayName = descriptors[i]!!.getNativeName()
+            val tag = descriptors[i]!!.getTag()
+            Log.v(LOG_TAG, "$displayName => $tag ")
+            listLocaleNameAndTag.add(Language(displayName = displayName, tag = tag, index = i + 1))
+        }
+        return listLocaleNameAndTag
+    }
+
+    /**
+     * Saves the current selected language tag
+     * @property languageTag the tag of the language
+     */
+    fun saveCurrentLanguageInSharePref(languageTag: String) {
+        val sharedPref = PreferenceManager.getDefaultSharedPreferences(context)
+        with(sharedPref.edit()) {
+            putString(context.getString(R.string.pref_key_locale), languageTag)
+            apply()
+        }
+    }
+
+    /**
+     * Not every locale we ship can be used on every device, due to
+     * font or rendering constraints.
+     *
+     *
+     * This method filters down the list before generating the descriptor array.
+     */
+    private fun getUsableLocales(): Array<LocaleDescriptor?> {
+        val shippingLocales = LocaleManager.getPackagedLocaleTags()
+        val initialCount: Int = shippingLocales.size
+        val locales: MutableSet<LocaleDescriptor> = HashSet(initialCount)
+        for (tag in shippingLocales) {
+            val descriptor = LocaleDescriptor(tag)
+            if (!descriptor.isUsable()) {
+                Log.w(LOG_TAG, "Skipping locale $tag on this device.")
```

Comment:
We do not use `Log` directly [at least in new code]. Instead we usually create a `Logger()` (from `support-base`). The logger allows us to have more control about what gets logged when (e.g. in Fenix for example we limit what we log in release builds).

Reply:
I made the changes.